pkg/defaults/catsrcHelpers.go

```diff
@@ -181,5 +206,21 @@ func AreCatsrcSpecsEqual(spec1 *olmv1alpha1.CatalogSourceSpec, spec2 *olmv1alpha
 	spec1Copy.Image = strings.ToLower(spec1Copy.Image)
 	spec2Copy.Image = strings.ToLower(spec2Copy.Image)
 
-	return reflect.DeepEqual(spec1Copy, spec2Copy)
+	deepComparison := reflect.DeepEqual(spec1Copy, spec2Copy)
+
+	if !strings.EqualFold(string(spec1.SourceType), string(spec2.SourceType)) ||
```

Comment:
According to the bug, the main issue is changes on `spec.updateStrategy`. So perhaps you can simply reset that field on both objects and continue to use `DeepEqual`. Having said that, comparing only the fields you want is prudent but it is ugly code-wise.